spinalcordtoolbox/scripts/sct_dice_coefficient.py

```diff
@@ -122,18 +123,26 @@ def main(argv=None):
     curdir = os.getcwd()
     os.chdir(tmp_dir)  # go to tmp directory
 
+    im_1 = Image(fname_input1)
+    im_2 = Image(fname_input2)
+
     if arguments.bin is not None:
         fname_input1_bin = add_suffix(fname_input1, '_bin')
-        run_proc(['sct_maths', '-i', fname_input1, '-bin', '0', '-o', fname_input1_bin])
+        im_1_bin = im_1.copy()
+        im_1_bin.data = binarize(im_1.data, 0)
+        im_1_bin.save(fname_input1_bin)
         fname_input1 = fname_input1_bin
+
         fname_input2_bin = add_suffix(fname_input2, '_bin')
-        run_proc(['sct_maths', '-i', fname_input2, '-bin', '0', '-o', fname_input2_bin])
+        im_2_bin = Image(fname_input2)
+        im_2_bin.data = binarize(im_2.data, 0)
+        im_2_bin.save(fname_input2_bin)
         fname_input2 = fname_input2_bin
 
     # copy header of im_1 to im_2
-    from spinalcordtoolbox.image import Image
     im_1 = Image(fname_input1)
     im_2 = Image(fname_input2)
```

Comment:
Nit: It feels a bit weird to see the `im_1 = Image(fname_input1)` and `im_2 = Image(fname_input2)` lines repeated twice. We're saving the images to disk, only to immediately reload them to the same variables.

If we perform the binarize steps in-place (i.e. modifying `im_1` and `im_2` directly), then I think we wouldn't have reload the images in order to copy the header over.

Reply:
agreed-- that also made me realize that we should probably merge the DICE calculation 👉 https://github.com/neuropoly/spinalcordtoolbox/issues/3201

Reply:
Yes definitely! Didn't pay attention :sweat_smile:

Reply:
Hmm, I'm not sure the fix in d4fed51 works how we want it to...

Right now `im_1` and `im_2` refer to the original image files, while  `im_1_bin` and `im_2_bin` refer to the binarized ones. So, right now, we save the header to the wrong image file if if `arguments.bin` is passed:

https://github.com/neuropoly/spinalcordtoolbox/blob/d4fed51aabe3f51a8c2f83a991cbd0fb4069f6ee/spinalcordtoolbox/scripts/sct_dice_coefficient.py#L142-L145

Reply:
Oups sorry about that, I'm a bit distracted. Do you agree with https://github.com/neuropoly/spinalcordtoolbox/commit/4652767faa1fbe09c52bcac6acc57a07fc5a19dc ?

Reply:
It looks good! Although, I just noticed another detail in the above snippet. :sweat_smile: 

There is both `im_1_bin = im_1.copy()` and `im_2_bin = Image(fname_input2)`...

Reply:
What do you think of this?

```python
    im_1 = Image(fname_input1)
    im_2 = Image(fname_input2)

    if arguments.bin is not None:
        fname_input1 = add_suffix(fname_input1, '_bin')
        im_1.data = binarize(im_1.data, 0)
        im_1.save(fname_input1, mutable=True)

        fname_input2 = add_suffix(fname_input2, '_bin')
        im_2.data = binarize(im_2.data, 0)
        im_2.save(fname_input2, mutable=True)
```

Reply:
Side note, but it is a bit weird that we have to update the fname in two places... 

1. In the image variables, using `mutable=True`
2. In the filename variables

Reply:
Sure. Save some RAM at the same time :+1: